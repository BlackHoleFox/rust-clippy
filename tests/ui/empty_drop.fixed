// run-rustfix
#![warn(clippy::empty_drop)]
#![allow(unused)]

// should cause an error
struct Foo;



// shouldn't cause an error
struct Bar;

impl Drop for Bar {
    fn drop(&mut self) {
        println!("dropping bar!");
    }
}

// should error
struct Baz;



<<<<<<< HEAD
fn main() {
    let s = Foo;
    // s goes out of scope.
}
=======
fn main() {}
>>>>>>> a4978d27e777f4ee7cbc8e2755238d5b3b957c3d
