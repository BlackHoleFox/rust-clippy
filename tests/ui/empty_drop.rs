// run-rustfix
#![warn(clippy::empty_drop)]

struct Foo;

impl Drop for Foo {
    fn drop(&mut self) {}
}

fn main() {
    let s = Foo;
    // s goes out of scope.
}
