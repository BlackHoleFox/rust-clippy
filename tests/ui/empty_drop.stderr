error: empty drop implementation
  --> $DIR/empty_drop.rs:5:1
   |
LL | / impl Drop for Foo {
LL | |     fn drop(&mut self) {}
LL | | }
   | |_^
   |
   = note: `-D clippy::empty-drop` implied by `-D warnings`
   = help: try removing this impl

error: aborting due to previous error

